dvc/state.py

```diff
@@ -455,7 +455,10 @@ def get(self, path_info):
             return None
 
         self._update_state_record_timestamp_for_inode(actual_inode)
-        return HashInfo("md5", value, size=int(actual_size))
+
+        mode = oct(os.stat(path).st_mode)
```

Comment:
Is full mode intentional? Or did you only mean to include permission bits?

Reply:
I'm leaning now more towards only saving user executable bit, so we could have a boolean parameter:

```
is_user_executable: false/true
```